src/main/java/io/debezium/connector/jdbc/type/debezium/ZonedTimeType.java

```diff
@@ -65,15 +68,33 @@ public void bind(Query<?> query, int index, Schema schema, Object value) {
             query.setParameter(index, null);
         }
         else if (value instanceof String) {
-            // NOTE:
-            // We must bind the value as an Instant as Hibernate will refuse to bind the TZ details
-            // on the destination correctly for Oracle.
-            final OffsetTime offsetTime = OffsetTime.parse((String) value, ZonedTime.FORMATTER);
-            query.setParameter(index, offsetTime.atDate(LocalDate.now()).toInstant());
+            final ZonedDateTime zdt = OffsetTime.parse((String) value, ZonedTime.FORMATTER).atDate(LocalDate.now()).toZonedDateTime();
+            if (getDialect().isJdbcTimeZoneSet()) {
+                query.setParameter(index, zdt, StandardBasicTypes.ZONED_DATE_TIME_WITH_TIMEZONE);
+            }
+            else {
+                if (getDialect().isConnectionTimeZoneSet()) {
+                    query.setParameter(index, zdt.withZoneSameInstant(getDatabaseTimeZone().toZoneId()));
+                }
+                else if (getDialect() instanceof SqlServerDatabaseDialect) {
```

Comment:
I think theis should be extracted into dialect itself if necessary.

Reply:
It is moved into a SQL Server Type implementation that extends and overrides the SQL Server-specific bits.